app/src/test/java/com/example/iphonesearchapi/viewmodel/detailViewModelTest.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.iphonesearchapi.viewmodel
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.example.iphonesearchapi.getOrAwaitValue
+import com.example.iphonesearchapi.model.ITunesResponse
+import com.example.iphonesearchapi.model.ResultOf
+import com.example.iphonesearchapi.network.IphoneApiService
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.*
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestRule
+import retrofit2.Response
+
+class detailViewModelTest {
+    @get:Rule
+    val testInstantTaskExecutorRule: TestRule = InstantTaskExecutorRule()
+    private val testDispatcher = TestCoroutineDispatcher()
+    private val testCoroutineScope = TestCoroutineScope(testDispatcher)
```

Comment:
We need to do this in every test file we create. We could extract this into a base test class and have all other test classes extend this base class.